### PR TITLE
Chore/envelopes monitor segmented messages

### LINKED ISSUE #4310: Adapt envelopes monitor to segmented messages
Interesting, we probably need to update at some point what it means for a message to be dispatched received?

_Originally posted by @cammellos in https://github.com/status-im/status-go/pull/4285#discussion_r1391249264_

### PR DESCRIPTION
**Pull request by osmaczko**
Adapts tracking for segmented messages.

closes: #4310

### CODE REVIEW

File: protocol/messenger_messages_tracking_test.go [hunk 1]
```diff
@@ -151,12 +151,13 @@ func (s *MessengerMessagesTrackingSuite) newMessenger(waku types.Waku, logger *z
 	return messenger, interceptor
 }
 
-func (s *MessengerMessagesTrackingSuite) TestMessageMarkedAsSent() {
+func (s *MessengerMessagesTrackingSuite) testMessageMarkedAsSent(textSize int) {
```
Comment (cammellos) [inline protocol/messenger_messages_tracking_test.go:154]: this looks disabled?

**Review by cammellos — APPROVED** [review]
looks good, thank you, left a couple of minor comments

File: protocol/messenger_messages_tracking_test.go (hunk 1 above)
Comment (osmaczko) [reply protocol/messenger_messages_tracking_test.go:154]: It is used by next two tests: 

```
func (s *MessengerMessagesTrackingSuite) TestMessageMarkedAsSent() {
	s.testMessageMarkedAsSent(1)
}
func (s *MessengerMessagesTrackingSuite) TestSegmentedMessageMarkedAsSent() {
	s.testMessageMarkedAsSent(4 * 1024 * 1024) // 4MB - ensure message is segmented
}
```

Comment (cammellos) [reply protocol/messenger_messages_tracking_test.go:154]: duh :facepalm:

**Review by jrainville — APPROVED** [review]
LGTM

**Pull request merged**